Source/CombatExtended/Harmony/Harmony_FloatMenuMakerMap.cs

```diff
@@ -417,4 +417,36 @@ static bool ForceWearInventoryCheck(Pawn pawn, Apparel apparel, List<FloatMenuOp
             return true;
         }
     }
+
+    //Make non-lethal weapons equippable by non-violent pawns
+    [HarmonyPatch(typeof(FloatMenuMakerMap), "AddHumanlikeOrders")]
+    class Harmony_FloatMenuMakerMap_AddHumanlikeOrders_Patch
+    {
+        [HarmonyTranspiler]
+        static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            bool patchApplied = false;
+            List<CodeInstruction> instructionsArray = instructions.ToList();
+            for (int i = 0; i < instructions.Count(); i++)
+            {
+                if (instructionsArray[i].opcode == OpCodes.Callvirt && ReferenceEquals(instructionsArray[i].operand, AccessTools.PropertyGetter(typeof(ThingDef), nameof(ThingDef.IsWeapon))))
+                {
+                    instructionsArray[i].opcode = OpCodes.Call;
+                    instructionsArray[i].operand = AccessTools.Method(typeof(Harmony_FloatMenuMakerMap_AddHumanlikeOrders_Patch), nameof(IsLethalWeapon));
+                    patchApplied = true;
+                    break;
+                }
+            }
+            if (!patchApplied)
+            {
+                Log.Error($"{MethodBase.GetCurrentMethod().DeclaringType} did not find opcode to apply transpiler patch.");
+            }
+            return instructionsArray;
+        }
```

Comment:
This whole thing can be simplified by using Harmony's `MethodReplacer` extension method. https://github.com/pardeike/Harmony/blob/master/Harmony/Public/Transpilers.cs#L19

```suggestion
    class Harmony_FloatMenuMakerMap_AddHumanlikeOrders_Patch
    {
        [HarmonyTranspiler]
        static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
        {
            return instructions.MethodReplacer(AccessTools.PropertyGetter(typeof(ThingDef), nameof(ThingDef.IsWeapon)), AccessTools.Method(typeof(Harmony_FloatMenuMakerMap_AddHumanlikeOrders_Patch), nameof(IsLethalWeapon)));
        }
```

Note that I have not remotely tested this. It Should™ Work™ but if you do go that route test it thoroughly.